rclpy/rclpy/impl/__init__.py

```diff
@@ -26,6 +26,18 @@ def _import(name):
         with add_dll_directories_from_env('PATH'):
             return importlib.import_module(name, package='rclpy')
     except ImportError as e:
+        if e.path is None:
+            import sysconfig
+            expected_path = os.path.join(
+                os.path.dirname(os.path.dirname(__file__)),
+                name[1:] + sysconfig.get_config_var('EXT_SUFFIX'))
+            assert not os.path.isfile(expected_path)
```

Comment:
can you explain what this assert is checking?
I don't get its purpose, neither what `expected_path` is.

Reply:
After rechecking, I got what `expected_path` is.

I think it would be better something like:
```python3
if os.path.is_file(expected_path):
    e.msg += ...
```

So if the file exist you get a normal import error, instead of an assert error.

Reply:
The assertion is only ensuring the logic of the code. If the file does exist this branch of the code should never be taken since `e.path` would not be `None`. Therefore it should stay an `assert`.

Reply:
sounds reasonable